scripts/install-macos.sh

```diff
@@ -0,0 +1,268 @@
+#!/bin/bash
+# shellcheck disable=SC2016
+set -e
+
+# Set this machine's dependencies status(es) with the OK flag
+OK=true;
+
+# Dependency test functions begin here
+
+function isNode() {
+    if node -v; then
+        OK=true;
+    else
+        OK=false;
+    fi
+}
+
+function isYarn() {
+    if yarn --version; then
+        OK=true;    
+    else
+        OK=false;
+    fi
+}
+
+function isGit() {
+    if git --version; then
+        OK=true;
+    else
+        OK=false;
+    fi
+}
+
+function isGo() {
+    if go version; then
+        OK=true;
+    else
+        OK=false;
+    fi
+}
+
+function isGlide() {
+    if glide --version; then
+        OK=true;
+    else
+        OK=false;
+    fi
+}
+
+# Installation functions begin here
+
+function installGo() {
+    echo 'Installing Golang on your machine.'
+    # Go install script based on https://github.com/canha/golang-tools-install-script
+    VERSION="1.15.7"
+
+    [ -z "$GOROOT" ] && GOROOT="$HOME/go/go-install-do-not-delete"
+    [ -z "$GOPATH" ] && GOPATH="$HOME/go"
+
+    OS="$(uname -s)"
+    ARCH="$(uname -m)"
+
+    case $OS in
+        "Linux")
+            case $ARCH in
+            "x86_64")
+                ARCH=amd64\
+                ;;
+            "aarch64")
+                ARCH=arm64
+                ;;
+            "armv6" | "armv7l")
+                ARCH=armv6l
+                ;;
+            "armv8")
+                ARCH=arm64
+                ;;
+            .*386.*)
+                ARCH=386
+                ;;
+            esac
+            PLATFORM="linux-$ARCH"
+        ;;
+        "Darwin")
+            PLATFORM="darwin-amd64"
+        ;;
+    esac
+
+    if [ -z "$PLATFORM" ]; then
+        echo "Your operating system is not supported by the script."
```

Comment:
can we add the name of the OS that is unsupported?

`echo "Your operating system '$OS' is not supported by the script."`